### ISSUE TITLE
Rack::Static middleware works with and without root url in urls parameter

### ISSUE THREAD

**Issue by akonsu** [rack/rack#740]
this is originally posted on stackoverflow: http://stackoverflow.com/questions/26167068/my-rackstatic-middleware-works-with-root-url-in-urls-parameter-and-without
I have this in my `config.ru`:

```
require 'rack'
require './app.rb'

use(Rack::Static,
  :urls => %w(/css /fonts /images /js),
  :root => '../static',
  :index => 'index.html')
end

  run MyApp
```

it correctly serves static files for all four of the above URLs, but in addition, it also serves `index.html` from the root URL as if I had `:urls => %w(/ /css /fonts /images /js)`.

**Comment by mcantor** [comment]
For the sake of clarity, please confirm I have all of this correct:

Your local filesystem looks like this:

```
├── static/
│   ├── css/
│   ├── fonts/
│   ├── images/
│   ├── js/
│   └── index.html
└── app/
    ├── config.ru
    └── app.rb
```

# Expected Behavior

Request  | Response
------------------- | ---------------
`GET / `           | Content of `static/index.html`
`GET /index.html` | `404 Not Found`

# Actual Behavior

Request  | Response
------------------- | ---------------
`GET / `           | ???
`GET /index.html` | Content of `static/index.html`

Is that correct?

**Comment by mig-hub** [comment]
I did not test it yet but it looks like it could because it tests for `route_file || override_file_path`, and `override_file_path` only tests against this regexp `/\/$/`. So it looks like it `can_serve` paths which do not start with the URLs, but end with a forward slash.

Line 99 is not easy to understand for me.

**Comment by jeremyevans** [comment]
This should be fixed in 2.1.0.
